File: src/main/java/com/github/thorben/webnd/domain/user/EmailAddress.java
package com.github.thorben.webnd.domain.user;

import com.github.thorben.webnd.foundation.AbstractValueObject;

import javax.persistence.Embeddable;

@Embeddable
public class EmailAddress extends AbstractValueObject<String> {

	protected EmailAddress() {
		// JPA
	}

	public EmailAddress(String value) {
		super(value);
	}
}

File: src/main/kotlin/com/github/thorben/webnd/domain/user/EmailAddress.kt
package com.github.thorben.webnd.domain.user

import com.github.thorben.webnd.foundation.AbstractValueObject
import javax.persistence.Embeddable

@Embeddable
class EmailAddress(value: String) : AbstractValueObject<String>(value) {
	protected constructor(): this("") {
		// JPA
	}
}